Improve CLI config documentation (#8130)
doctrine/doctrine2: docs/en/reference/tools.rst

~~~~~~~~~~~~~

Whenever the ``doctrine`` command line tool is invoked, it can
access all Commands that were registered by a developer. There is no
auto-detection mechanism at work. The Doctrine binary
already registers all the commands that currently ship with
Doctrine DBAL and ORM. If you want to use additional commands you
have to register them yourself.

All the commands of the Doctrine Console require access to the
``EntityManager`` or the ``DBAL`` Connection. You have to inject them into

> the console application using so called Helper-Sets. Whenever you invoke

Reviewer: Why use capitals here? If these refers to a class or interface, it should be quoted.

> the Doctrine binary the current folder is searched for the file

Reviewer: ```suggestion
the Doctrine binary the current directory is searched for the
```

``cli-config.php`` file. This file contains the project specific
configuration.

> If your application uses the ORM package, the project-specifc

Reviewer: ```suggestion
If your application uses the ORM package, the project-specific
```

``cli-config.php`` looks like this:

.. code-block:: php

    <?php
    use Doctrine\ORM\Tools\Console\ConsoleRunner;

>     // replace with file to your own project bootstrap

Reviewer 2: ```suggestion
    // replace this with the path to your own project bootstrap file.
```

    require_once 'bootstrap.php';

    // replace with mechanism to retrieve EntityManager in your app
    $entityManager = GetEntityManager();

    return ConsoleRunner::createHelperSet($entityManager);

> If your application uses the `DBAL` connection only, you construct the

Reviewer: ```suggestion
If your application uses the ``DBAL`` connection only, you construct the
```

helper set like this:

.. code-block:: php

    <?php
    use Doctrine\DBAL\Tools\Console as DBALConsole;
    use Symfony\Component\Console\Helper\HelperSet;

    // replace with file to your own project bootstrap
    require_once 'bootstrap.php';

    // replace with mechanism to retrieve EntityManager in your app
    $connection = GetDbalConnection();

    return new HelperSet([
        'db' => new DBALConsole\Helper\ConnectionHelper($connection),

>     ]);

Reviewer 2: I don't see a reason why the ORM docs have a "If your application uses the `DBAL` connection only" part. I think this can be removed from the ORM docs.

Author: It's my understanding that migrations can be used with and without the ORM (the latter with some additional code and effort). There is the chapter "[Diffing without the ORM](https://www.doctrine-project.org/projects/doctrine-migrations/en/2.2/reference/generating-migrations.html#diffing-without-the-orm)".

Reviewer 2: Yes and I wouldn't expect "without the ORM" to be found in the ORM documentation. :-)
It is more fitting to find such an information in the Migrations documentation than in ORM and it actually can be found in "[Custom Integration](https://www.doctrine-project.org/projects/doctrine-migrations/en/2.2/reference/custom-integration.html)".
